### PR TITLE
add : like functionality and single video page

### PR DESCRIPTION
**Pull request by ApurvaSawant11**
In this PR, I have added Like functionality and single video page

### Files to check:
- Like and dropdown context
- SingleVideo.jsx
- LikedVideos.jsx

### Files to ignore:
- files containing minor changes

### Note:  Click on video image to navigate to single video page and test the like functionality

### CODE REVIEW

File: src/components/card/listCard/ListCard.jsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import React from "react";
+import { MoreIcon } from "../../../assets";
+import "./listCard.css";
+import { useAuth, useDropdown } from "../../../context";
```
Comment (Dev-AkshatGupta) [inline src/components/card/listCard/ListCard.jsx:4]: Suggestion : Use absolute import much better way to get things done

File: src/components/card/listCard/ListCard.jsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+const ListCard = ({ video }) => {
+  const { token } = useAuth();
+  const { _id, title, isInWatchLater, isInLiked } = video;
+  const { showDropdown, toggleShowDropdownList, getDropdownList } =
+    useDropdown();
```
Comment (Dev-AkshatGupta) [inline src/components/card/listCard/ListCard.jsx:10]: Nice naming apoorva

File: src/components/card/listCard/listCard.css [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    width: 30rem;
+}
+
+.dropdown-icon {
+    align-self: center;
+    cursor: pointer;
+}
+
+.list-dropdown {
+    /* right:0rem; */
```
Comment (Dev-AkshatGupta) [inline src/components/card/listCard/listCard.css]: suggestion : Make sure before your submission of removing commented codes

File: src/context/dropdown-context.js [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+        ? [
+            {
+              _id: uuid(),
+              option: "Remove from History",
+              Icon: () => <TrashIcon size={24} className="mr-0p5" />,
+            },
+          ]
+        : dropdownList;
+
+    // dropdown for pages other than home and history
```
Comment (Dev-AkshatGupta) [inline src/context/dropdown-context.js:66]: I liked the comments they some how increase the readability and understandability of the code

**Review by Dev-AkshatGupta — APPROVED** [review]
Everything seems good to e I have mentioned some comments nothing specific just suggestions
You are good to merge with this even

**Review by Hamza-777 — APPROVED** [review]
Everything looks good, looks good to merge.

File: src/components/card/listCard/ListCard.jsx (hunk 1 above)
Comment (ApurvaSawant11) [reply src/components/card/listCard/ListCard.jsx:4]: Okay. Will refactor in coming PRs

File: src/components/card/listCard/listCard.css (hunk 3 above)
Comment (ApurvaSawant11) [reply src/components/card/listCard/listCard.css]: I have removed it. Thanks for pointing out

**Pull request merged**